temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -246,9 +252,70 @@ private StartWorkflowExecutionRequest overrideStartWorkflowExecutionRequest(
     if (taskTimeoutMillis != Durations.toMillis(request.getWorkflowTaskTimeout())) {
       request.setWorkflowTaskTimeout(Durations.fromMillis(taskTimeoutMillis));
     }
+
+    SearchAttributes.Builder searchAttributes = request.getSearchAttributes().toBuilder();
+    Map<String, Payload> payloadMap = request.getSearchAttributes().getIndexedFieldsMap();
+    for (Map.Entry<String, Payload> searchAttribute : payloadMap.entrySet()) {
+      String key = searchAttribute.getKey();
+      SearchAttributeType type = searchAttributeToType(key);
+      if (type != null && !SearchAttributeType.Unspecified.name().equals(type.name())) {
+        searchAttributes.putIndexedFields(
+            key,
+            searchAttribute
+                .getValue()
+                .toBuilder()
+                .putMetadata(
+                    EncodingKeys.METADATA_ENCODING_KEY, EncodingKeys.METADATA_ENCODING_JSON)
+                .putMetadata(EncodingKeys.METADATA_TYPE_KEY, ByteString.copyFromUtf8(type.name()))
+                .build());
+      } else {
+        searchAttributes.removeIndexedFields(key);
+        throw Status.INVALID_ARGUMENT.asRuntimeException();
+      }
+    }
+    request.setSearchAttributes(searchAttributes);
     return request.build();
   }
 
+  private SearchAttributeType searchAttributeToType(String key) {
```

Comment:
should it be called systemSearchAttributeToType? It looks like it works only with our system/default attributes

Reply:
Custom keys are also there.

Reply:
@vkoby that's exactly the problem